android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -308,6 +308,52 @@ private static void initSystemKeyboard(Context appContext) {
     }
   }
 
+  /*
+  // TODO: Chromium has a bug where deleteSurroundingText deletes an entire grapheme cluster
+  // instead of one code-point.
+  // We'll retrieve up to (dn*2+16) characters before the cursor to collect enough characters
+  // for surrogate pairs + a long grapheme cluster.
+  // This buffer will be used to put back characters as-needed
+  //
+  // Visibility public so we can run a unit test
+  */
+  public static void performLeftDeletions(InputConnection ic, int dn) {
```

Comment:
We currently can't run input connections in a unit test, so I intend to make this private